stats/statsd.go

```diff
@@ -159,18 +159,24 @@ func (s *statsdStats) internalNewTaggedStat(name, statType string, tags Tags, sa
 	}
 
 	// Clean up tags based on deployment type. No need to send workspace id tag for free tier customers.
-	for excludedTag := range s.config.excludedTags {
-		delete(tags, excludedTag)
-	}
-	if tags == nil {
-		tags = make(Tags)
-	}
-	if v, ok := tags[""]; ok {
-		s.logger.Warnf("removing empty tag key with value %s for measurement %s", v, name)
-		delete(tags, "")
+	newTags := make(Tags)
+	for k, v := range tags {
+		if strings.Trim(k, " ") == "" {
+			s.logger.Warnf("removing empty tag key with value %q for measurement %q", v, name)
+			continue
+		}
+		if _, ok := s.config.excludedTags[k]; ok {
+			continue
+		}
```

Comment:
I thought I'd leave it for backwards compatibility in case somewhere there is an exclusion rule that would work only before the tag is sanitized. I thought it wouldn't hurt having for now.